src/mcpd/mcpd_client.py

```diff
@@ -431,24 +443,57 @@ def agent_tools(self, servers: list[str] | None = None) -> list[Callable[..., An
             but may not be suitable for pickling due to the embedded client state.
         """
         agent_tools = []
-        all_tools = self.tools()
 
         # Determine which servers to use.
-        servers_to_use = all_tools.keys() if servers is None else servers
+        servers_to_use = self.servers() if servers is None else servers
 
-        # Fetch tools from selected servers.
+        # Filter to healthy servers if requested (one HTTP call for all servers).
+        if check_health:
+            servers_to_use = self._get_healthy_servers(servers_to_use)
+
+        # Fetch tools from selected servers only (avoids fetching from unhealthy servers).
         for server_name in servers_to_use:
-            if server_name not in all_tools:
-                # Server doesn't exist or has no tools - skip silently.
+            try:
+                tool_schemas = self.tools(server_name=server_name)
+            except (ServerNotFoundError, ServerUnhealthyError):
+                # Server doesn't exist or became unhealthy - skip silently.
```

Comment:
I don't really like skipping this silently. We could, at least, log a warning that we're skipping this server because it's unhealthy. This way the user could look into the logs and find out why they can't connect to a specific server. Maybe this is something to do in the `_get_healthy_servers` method though.

Reply:
Yep, @daavoo also mentioned this and it's going to be covered in a PR shortly for Python and JS SDKs 🙏🏼 

See: https://github.com/mozilla-ai/mcpd-sdk-python/pull/31#discussion_r2524828756